enterprise/glassfish.common/src/org/netbeans/modules/glassfish/common/LogViewMgr.java

```diff
@@ -1196,27 +1197,25 @@ private static FetchLog getServerLogStream(
             final GlassfishInstance instance) {
         FetchLog log;
         FetchLog deadLog = null;
-        synchronized (serverInputStreams) {
-            log = serverInputStreams.get(instance);
-            if (log != null) {
-                if (log instanceof FetchLogPiped) {
-                    // Log reading task in running state
-                    if (((FetchLogPiped) log).isRunning()) {
-                        return log;
-                    // Log reading task is dead
-                    } else {
-                        // Postpone cleanup after synchronized block.
-                        deadLog = log;
-                        removeLog(instance);
-                    }
-                } else {
+        log = serverInputStreams.computeIfPresent(instance, (key, value) -> value);
```

Comment:
Again changed semantics. Code that was atomic before is not anymore.

Reply:
Again, you are correct. Thanks for the observations.